src/AnnotationsServiceProvider.php

```diff
@@ -70,6 +73,13 @@ class AnnotationsServiceProvider extends ServiceProvider
      */
     protected $scanEverything = false;
 
+    /**
+     * Determines whether to use attributes for scanning.
+     *
+     * @var bool
+     */
+    protected $useAttribute = false;
```

Comment:
this could be `$attributeStrategy = RouteAttributesScanner::class` and this way you don't have to use `if` in your service provider factories to check which one do you want to use...